src/main/java/org/spongepowered/api/tag/TagType.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of SpongeAPI, licensed under the MIT License (MIT).
+ *
+ * Copyright (c) SpongePowered <https://www.spongepowered.org>
+ * Copyright (c) contributors
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package org.spongepowered.api.tag;
+
+import org.spongepowered.api.registry.RegistryType;
+import org.spongepowered.api.util.annotation.CatalogedBy;
+
+@CatalogedBy(TagTypes.class)
+public interface TagType<T extends Taggable> {
+
+    /**
+     * The id of this tag type.
+     */
+    String id();
```

Comment:
Is this just the ID that this tag type is registered with under the appropriate registry? What does this ID look like? _Why_ do we need this to create tag types?

Reply:
No - Its effectively a "key" that doesn't have a namespace. Its used to the folders in vanilla:
e.g `blocks` `entity_types`, `items`. In the registry they are registered under different keys.
These need to comply with minecraft's folder system - cannot be namespaced.
I thought it was best to call them `id`s otherwise doing a `Path` assumes that its stored in a file structure.
Open to suggestions though.

Reply:
For clarity, what would be the filepath of a self-added tag?

That will help determine if really necessary.

Reply:
datapackname/data/namespace/tags/tagtype (the id)/xxx.json

Datapacks / plugins are within their rights to add json files under namespaces other than their own.
Infact I have just realised that it is probably a good idea to have seperate datapacks per plugin. Otherwise you're won't be able to specify the priority of datapacks, or disable individual datapacks, and possibly your checking of whether a datapack is still valid may be wrong too.

Reply:
Let's take it from a different angle.

I don't care if this is in the implementation as an implementation detail. Why is this in the API?

Reply:
@dualspiral I see, in the case if someone was to add their own `TagType`...basically if someone was to make something else `Taggable`. It just so happens to correlate to the sub directory under `tags`

That said, @tyhdefu I don't think we can support this. While this is ultimately an implementation detail, the client will die if it gets unknown types sent through the protocol. I...think, perhaps I am wrong.

and maybe not able to support isn't entirely right...a system would need to go into place for "logic"/"server-only" tag types.

Reply:
Removed from the API.